assigner.py

```diff
@@ -20,173 +22,257 @@
+
     with config(args.config) as conf:
+        host = conf['gitlab-host']
+        namespace = conf['namespace']
+        token = conf['token']
+        roster = conf['roster']
+        semester = conf['semester']
+
         count = 0
-        for student in conf['roster']:
-            if args.section and student['section'] != args.section:
+        for student in roster:
+            username = student['username']
+            s_section = student['section']
+
+            if section and s_section != section:
                 continue
 
-            name = StudentRepo.name(conf['semester'], student['section'], args.name, student['username'])
+            full_name = StudentRepo.name(semester, s_section, hw_name, username)
 
             try:
-                repo = StudentRepo(conf['gitlab-host'], conf['namespace'], name, conf['token'])
+                repo = StudentRepo(host, namespace, full_name, token)
                 if 'id' not in student:
-                    student['id'] = Repo.get_user_id(student['username'], conf['gitlab-host'], conf['token'])
+                    student['id'] = Repo.get_user_id(username, host, token)
 
                 repo.add_member(student['id'], Access.developer)
                 count += 1
             except RepoError:
-                logging.warn("Could not add %s to %s", student['username'], name)
+                logging.warn("Could not add {} to {}.".format(username, full_name))
             except HTTPError as e:
                 raise
                 if e.response.status_code == 404:
-                    logging.warn("Repository %s does not exist.", name)
+                    logging.warn("Repository {} does not exist.".format(full_name))
                 else:
                     raise
 
-    print("Granted access to ", count, " repositories")
+    print("Granted access to {} repositories.".format(count))
 
 
 def get(args):
-    if args.student:
+    """Creates a folder for the assignment in the CWD (or <path>, if specified)
+    and clones each students' repository into subfolders.
+    """
+    hw_name = args.name
+    hw_path = args.path
+    section = args.section
+    target = args.student  # used if assigning to a single student
+
+    if target:
         raise NotImplementedError("'--student' is not implemented")
 
     with config(args.config) as conf:
-        path = os.path.join(args.path, args.name)
+        host = conf['gitlab-host']
+        namespace = conf['namespace']
+        token = conf['token']
+        roster = conf['roster']
+        semester = conf['semester']
+        path = os.path.join(hw_path, hw_name)
         os.makedirs(path, mode=0o700, exist_ok=True)
 
         count = 0
-        for student in conf['roster']:
-            if args.section and student['section'] != args.section:
+        for student in roster:
+            username = student['username']
+            s_section = student['section']
+
+            if section and s_section != section:
```

Comment:
Again with un-hacking...I should probably do that myself since it's my fault in the first place.